kotlin/README.md

```diff
@@ -0,0 +1,616 @@
+    } 
+}
+```
+
+When declaring parameter names in a multiline lambda, put the names on the first line, 
+followed by the arrow and the newline:
+
+```
+appendCommaSeparated(properties) { 
+    prop -> val propertyValue = prop.get(obj)  // ... 
+}
+```
+
+If the parameter list is too long to Dt on a line, put the arrow on a separate line:
+
+```
+foo { 
+    context: Context, 
+    environment: Env 
+    -> 
+    context.configureEnv(environment) 
+}
+```
+
+### Documentation comments:
+For longer documentation comments, place the opening /** on a separate line and begin 
+each subsequent line with an asterisk:
+
+```
+/** 
+* This is a documentation comment 
+* on multiple lines. 
+*/
+```
+
+Short comments can be placed on a single line:
+
+```
+/** This is a short documentation comment. */
+```
+
+String templates:
+Don't use curly braces when inserting a simple variable into a string template. 
+Use curly braces only for longer expressions. 
+
+```
+println("$name has ${children.size} children")
+```
+
+### Type aliases:
+If you have a functional type or a type with type parameters which is used multiple times 
+in a codebase, prefer deDning a type alias for it: 
+
+```
+typealias PersonIndex = Map<String, Person>
+```
+
+### Using conditional statements:
+Prefer using the expression form of try, if and when. Examples:
+
+```
+return if (x) foo() else bar()
+
+return when(x) { 
+    0 -> "zero" 
+    else -> "nonzero" 
+}
+```
+
+The above is preferable to:
+
+```
+if (x) 
+    return foo()
+else 
+    return bar() 
+
+when(x) { 
+    0 -> return "zero" 
+    else -> return "nonzero" 
+}
+```
+
+#### if versus when
+Prefer using if for binary conditions instead of when . Instead of
+
+```
+when (x) { 
+    y -> // ... 
+    else -> // ... 
+}
+```
+
+use
+
+```
+if (x == y) ... else …
+```
+
+Prefer using when if there are three or more options.
+
+Loops on ranges:
+Use the until function to loop over an open range:
+
+```
+for (i in 0..n - 1) { /*...*/ } // bad 
+for (i in 0 until n) { /*...*/ } // good
+```
+
+### Using extension functions:
+Use extension functions liberally. Every time you have a function that works primarily 
+on an object, consider making it an extension function accepting that object as a receiver.
+To minimize API pollution, restrict the visibility of extension functions as much as it 
+makes sense. As necessary, use local extension functions, member extension functions, 
+or top-level extension functions with private visibility.
+
+### Using scope functions apply/with/run/also/let:
+Kotlin provides a variety of functions to execute a block of code in the context of a 
+given object: let, run, with, apply, and also. You should be used the scope functions 
+depending on your needs, there is not a specific rule for each one, they are used in 
+order to have a clean and easy to maintain code, and since Kotlin is a compiled language
+with the use of those functions you can get better performance of your apps.
```

Comment:
I think that it looks better like this:

Kotlin provides a variety of functions to execute a block of code in the context of a given object: let, run, with, apply, and also. You should use the scope functions depending on your needs. There is not a specific rule for each one, they are used to have clean and easy to maintain code, and since Kotlin is a compiled language, with the use of those functions, you can get better performance of your apps.